Mathlib/Data/Finsupp/Defs.lean

```diff
@@ -1034,6 +1034,52 @@ theorem induction_linear {p : (α →₀ M) → Prop} (f : α →₀ M) (h0 : p
     (hadd : ∀ f g : α →₀ M, p f → p g → p (f + g)) (hsingle : ∀ a b, p (single a b)) : p f :=
   induction₂ f h0 fun _a _b _f _ _ w => hadd _ _ w (hsingle _ _)
 
+section LinearOrder
+
+variable [LinearOrder α] {p : (α →₀ M) → Prop}
+
+/-- A finitely supported function can be built by adding up `single a b` for increasing `a`. -/
+theorem induction_on_max (f : α →₀ M) (h0 : p 0)
+    (ha : ∀ (a b) (f : α →₀ M), (∀ c ∈ f.support, c < a) → b ≠ 0 → p f → p (single a b + f)) :
```

Comment:
Would it make sense to have a version for partial orders with `∀ c ∈ f.support, c < a` replaced by `∀ c ∈ f.support, ¬ a ≤ c`? I think that could also be useful.

Reply:
Another thing is that it would be nice to eliminate to `Sort*`
But maybe both of these fall outside the scope of this PR?

Reply:
This proof relies mostly on `Finset.induction_on_max`, so these changes would have to be implemented there first.

Note that we need a linear order for the `Sort*` version to be useful, as otherwise whichever of two incomparable elements we build first would be entirely arbitrary.

Reply:
Ok fair. Let's keep it like this.